int/api/wallet/import.go

```diff
@@ -59,13 +59,34 @@ func (c *wImport) Handle(params operations.MgmtWalletImportParams) middleware.Re
 			})
 	}
 
-	newWallet, err := wallet.Imported(walletName, privateKey)
-	if err != nil {
-		return operations.NewMgmtWalletCreateInternalServerError().WithPayload(
-			&models.Error{
-				Code:    errorCodeWalletCreateNew,
-				Message: err.Error(),
-			})
+	newWallet, RequestError := wallet.Imported(walletName, privateKey)
+	if RequestError.Err != nil {
+		switch {
+		case RequestError.StatusCode == wallet.ImportedAlreadyImported:
+			return operations.NewMgmtWalletCreateInternalServerError().WithPayload(
+				&models.Error{
+					Code:    errorCodeWalletAlreadyImported,
+					Message: RequestError.Err.Error(),
+				})
+		case RequestError.StatusCode == wallet.ImportedEncodingB58Error:
+			return operations.NewMgmtWalletCreateInternalServerError().WithPayload(
+				&models.Error{
+					Code:    errorCodeWalletEncodingB58E,
+					Message: RequestError.Err.Error(),
+				})
+		case RequestError.StatusCode == wallet.ImportedLoadingWalletsError:
+			return operations.NewMgmtWalletCreateInternalServerError().WithPayload(
+				&models.Error{
+					Code:    errorCodeWalletLoadingWallets,
+					Message: RequestError.Err.Error(),
+				})
+		default:
+			return operations.NewMgmtWalletCreateInternalServerError().WithPayload(
+				&models.Error{
+					Code:    errorCodeWalletCreateNew,
+					Message: RequestError.Err.Error(),
+				})
+		}
```

Comment:
I have the feeling here that we duplicate a lot of code. I think that `wallet.Imported` should return the error code or nil.

```suggestion
		return operations.NewMgmtWalletCreateInternalServerError().WithPayload(
				&models.Error{
					Code:    returnedErrorCode,
					Message: RequestError.Err.Error(),
				})
```

Reply:
okay, is there a way that `wallet.Imported` returns `operations.NewMgmtWalletCreateInternalServerError`?

Reply:
@Thykof it lead us to the first issue : cyclic import

Reply:
so we cannot import just te errors which will be used by operations.NewMgmtWalletCreateInternalServerError
![image](https://user-images.githubusercontent.com/16125633/204545563-56c35ac7-28d6-4ad0-bb4b-cbc28900582d.png)

Reply:
@gregLibert do you have a better suggestion to avoid cycle import errors ? i used a custom errors inside pkg/wallet/wallet.go and a switch to match my return errors in int/api/wallet/errors.go since i cannot import int/api/wallet/errors.go inside pkg/wallet. but it seems its not very optimal solution

Reply:
is it possible to move errors.go to solve cyclic import?